### PR TITLE
lib/subsys: enable offloading posix appl compilation without NET_SOCKETS_POSIX_NAMES

### PR DESCRIPTION
**Pull request by jhirsi**
Enable POSIX_API application compilation with offloading/libmodem, i.e. without CONFIG_NET_SOCKETS_POSIX_NAMES for certain lib/subsys conponents.
Jira: MOSH-72
Signed-off-by: Jani Hirsimäki <jani.hirsimaki@nordicsemi.no>

### CODE REVIEW

File: lib/at_cmd/at_cmd.c [hunk 1]
```diff
@@ -7,7 +7,12 @@
 #include <logging/log.h>
 #include <zephyr.h>
 #include <stdio.h>
+#if !defined(CONFIG_NET_SOCKETS_POSIX_NAMES)
```
Comment (rlubos) [inline lib/at_cmd/at_cmd.c:10]: I'd rather use `#if defined(CONFIG_POSIX_API)` as a factor to decide whether to use POSIX headers or not. Same in other places below.

File: lib/nrf_modem_lib/nrf91_sockets.c [hunk 2]
```diff
@@ -247,13 +253,15 @@ static int z_to_nrf_flags(int z_flags)
 		nrf_flags |= NRF_MSG_PEEK;
 	}
 
+#if defined (CONFIG_NET_SOCKETS_POSIX_NAMES)
```
Comment (rlubos) [inline lib/nrf_modem_lib/nrf91_sockets.c:256]: We should rather add these flags to an appropriate POSIX header in sdk-zephyr instead of disabling them.

File: subsys/net/lib/download_client/src/download_client.c [hunk 3]
```diff
@@ -65,8 +72,11 @@ static int socket_timeout_set(int fd, int type)
 		.tv_usec = (timeout_ms % 1000) * 1000,
 	};
 
+#if !defined (CONFIG_NET_SOCKETS_POSIX_NAMES)
+	LOG_INF("Configuring socket timeout (%d s)", (int32_t)timeo.tv_sec);
```
Comment (rlubos) [inline subsys/net/lib/download_client/src/download_client.c:76]: What's the reason for this change, a different underlying type of `timeo.tv_sec` in POSIX/non-POSIX variant? IMO we could stick to a single approach with the cast, it should work in both cases.

Comment (jhirsi) [reply subsys/net/lib/download_client/src/download_client.c:76]: yes, at least with POSIX_API it is 64bit and original code resulted a warning: 
/home/jani/code_wa/ncs/nrf/subsys/net/lib/download_client/src/download_client.c:78:10: warning: format '%ld' expects argument of type 'long int', but argument 2 has type 'time_t' {aka 'long long int'} 

If remember right, casting was needed, %lld printing wasn't working. And a flagging it was a safe choice because didn't want to change the original.

File: lib/at_cmd/at_cmd.c (hunk 1 above)
Comment (jhirsi) [reply lib/at_cmd/at_cmd.c:10]: Yeah, that was actually my initial solution but then (don't remember why and from where ;-)) I came to conclusion that there might be a situation where posix headers are used without all stuff from POSIX_API -config. Not sure anymore if that is the real use case of these configs.

But OK, I'll change those under POSIX_API flags.

File: lib/nrf_modem_lib/nrf91_sockets.c (hunk 2 above)
Comment (jhirsi) [reply lib/nrf_modem_lib/nrf91_sockets.c:256]: Yes, agree, but that would be a zephyr change. So, until that, for this PR, we need to go with POSIX_API flagging here?

Comment (rlubos) [reply lib/nrf_modem_lib/nrf91_sockets.c:256]: You could post a PR towards `sdk-zephyr` and update manifest in this PR, we already have quite a few downstream changes in the networking headers:
https://github.com/nrfconnect/sdk-zephyr/commit/38a90a7768511cbe769d04ce827e160d8fa554b8
Similar changes could be applied to POSIX headers (symbols unsupported upstream).

**Review by sigvartmh — APPROVED** [review]
dfu_target changes looks okei

File: lib/nrf_modem_lib/nrf91_sockets.c (hunk 2 above)
Comment (jhirsi) [reply lib/nrf_modem_lib/nrf91_sockets.c:256]: OK, created PR to sdk-zephyr for adding those sckt flags to posix socket header: https://github.com/nrfconnect/sdk-zephyr/pull/476

I'll update the manifest into this PR when that is merged.

Comment (rlubos) [reply lib/nrf_modem_lib/nrf91_sockets.c:256]: Thanks, just one more request, please update the `sdk-zephyr` revision to `pull/476/head` already in the manifest, we want a CI run over that PR.

Comment (jhirsi) [reply lib/nrf_modem_lib/nrf91_sockets.c]: Thanks @rlubos. All should be now done in both related PRes.

**Comment by rlubos** [comment]
Could you reabase? The CI seems to be stuck due to conflict. Otherwise looks good.

File: subsys/net/lib/download_client/src/download_client.c [hunk 4]
```diff
@@ -65,8 +72,11 @@ static int socket_timeout_set(int fd, int type)
 		.tv_usec = (timeout_ms % 1000) * 1000,
 	};
 
+#if defined (CONFIG_POSIX_API)
```
Comment (rlubos) [inline subsys/net/lib/download_client/src/download_client.c:75]: From the checkpatch report:
```suggestion
#if defined(CONFIG_POSIX_API)
```

Comment (jhirsi) [reply subsys/net/lib/download_client/src/download_client.c]: fixed

File: west.yml [hunk 5]
```diff
@@ -52,7 +52,7 @@ manifest:
     # https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/modules.html
     - name: zephyr
       repo-path: sdk-zephyr
-      revision: 5acae16d9ea808785279b1de1cd177108c1b1354
+      revision: pull/476/head
```
Comment (rlubos) [inline west.yml:55]: Good news, we've managed to get `sdk-zephyr` PR before the upmerge, please update SHA :
`revision: 37430f7c6b7b1e075fff26961e8dcfab12b6c800`

Comment (jhirsi) [reply west.yml]: done. thanks!

**Review by rlubos — APPROVED** [review]
